.github/workflows/nightly_release.yml

```diff
@@ -0,0 +1,67 @@
+name: SalesforceCommerceCloud/pwa-kit/nightly_release
+on:
+  workflow_dispatch:
+  schedule:
+    # Run every day at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * *"
+
+jobs:
+  create_nightly_release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get Timestamp
+        run: |-
+          echo "release_timestamp=$(date +'%m%d%Y%H%M%S')" >> "$GITHUB_ENV"
+
+      - name: Get monorepo version
+        run: |-
+          version=`cat package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "monorepo_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get retail-react-app version
+        run: |-
+          version=`cat packages/template-retail-react-app/package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "retail_app_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get commerce-sdk-react version
+        run: |-
+          version=`cat packages/commerce-sdk-react/package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "commerce_sdk_react_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+
+      - name: Install Monorepo Dependencies
+        run: |-
+          # Install node dependencies
+          node ./scripts/gtime.js monorepo_install npm ci
+
+      - name: Create Release Branch
+        run: git checkout -b release-${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}
+
+      - name: Bump version (monorepo)
+        run: |-
+          npm run bump-version -- "${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (retail-react-app)
+        run: |-
+          npm run bump-version:retail-react-app -- "${{ env.retail_app_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (commerce-sdk-react)
+        run: |-
+          npm run bump-version:commerce-sdk-react -- "${{ env.commerce_sdk_react_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Push branch to origin (Publish to NPM)
+        run: |-
+          git config --global user.name ${{ secrets.GIT_CONFIG_USERNAME }}
+          git config --global user.email ${{ secrets.GIT_CONFIG_EMAIL }}
+          git commit -am "Release ${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+          git push --set-upstream origin release-${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}
```

Comment:
Until now, the point of running testing in CI was to fail if we detect an error before releasing.
Once the package is released to NPM we do smoke testing manually.

Now with nightly releases, we still want to run tests in CI before releasing a nightly release.
But once the packages are released to NPM there's no longer anyone manually smoke testing what we released to NPM.
Should we add an extra step after a nightly release to run tests on the real published packages by nightly build, so we know we released something that works?

Reply:
I'm good with this as long as we keep the process automated and it's not a new manual step

Reply:
So nightlies are intended to be buggy. And anyone using nightlies needs to be aware of the fact that the changes are not tested and the release should not be used in production. I'll add this to the README to be explicit about it. Also, we need to remove all manual steps in the process. So we will be doing nightly releases without any smoke testing. If stuff breaks it was meant to be that way.

Reply:
^Also once pushed to a `release-` branch, the main workflow does run unit tests before publishing to NPM. That should be sufficient.